### PR TITLE
Remove `RuboCop::RSpec::Language::NodePattern`

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-rspec/issues/1832

Can be merged next major update.

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/cop/rspec/expect_actual.rb
Comment (pirj) [inline lib/rubocop/cop/rspec/expect_actual.rb:84]: This statement seems outdated. I think it can.

File: lib/rubocop/cop/rspec/expect_actual.rb
Comment (pirj) [inline lib/rubocop/cop/rspec/expect_actual.rb:83]: It feels that this note is telling “don’t waste time to refactor”. Do you think it makes sense to remove it?

**Review by pirj — APPROVED** [review]
Thank you!

And with your other changes, it feels that 3.0 is not far ahead 🙌

File: CHANGELOG.md
Comment (pirj) [inline CHANGELOG.md:5]: Why is 2.28.0 here twice?

**Review by pirj — APPROVED** [review]
Looks good, thanks!

Let’s wait a bit for the dust to settle on the extraction before merging.

**Comment by bquorning** [comment]
Should we create a `2-x-stable` branch to backport non-v3 changes to while we prepare for the upcoming v3 release?

**Review by bquorning — APPROVED** [review]

**Comment by bquorning** [comment]
@ydah Now that we have a separate `2-x-stable` branch, I think we are ready to merge this pull request after the conflicts are fixed.

**Comment by bquorning** [comment]
FYI specs are failing.

**Comment by ydah** [comment]
memo) needs: https://github.com/rubocop/rubocop-rspec/pull/1848

**Review by bquorning — APPROVED** [review]

**Pull request merged**
